homeassistant/components/matter/binary_sensor.py

```diff
@@ -161,4 +161,100 @@ def _update_from_device(self) -> None:
+    ),
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=MatterBinarySensorEntityDescription(
+            key="SmokeCoAlarmBatteryAlertSensor",
+            measurement_to_ha=lambda x: x
+            != clusters.SmokeCoAlarm.Enums.AlarmStateEnum.kNormal,
+            translation_key="battery_alert",
+            device_class=BinarySensorDeviceClass.BATTERY,
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+        entity_class=MatterBinarySensor,
+        required_attributes=(clusters.SmokeCoAlarm.Attributes.BatteryAlert,),
+    ),
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=MatterBinarySensorEntityDescription(
+            key="SmokeCoAlarmTestInProgressSensor",
+            translation_key="test_in_progress",
+            device_class=BinarySensorDeviceClass.RUNNING,
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+        entity_class=MatterBinarySensor,
+        required_attributes=(clusters.SmokeCoAlarm.Attributes.TestInProgress,),
+    ),
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=MatterBinarySensorEntityDescription(
+            key="SmokeCoAlarmHardwareFaultAlertSensor",
+            translation_key="hardware_fault",
+            device_class=BinarySensorDeviceClass.PROBLEM,
+            entity_category=EntityCategory.DIAGNOSTIC,
+        ),
+        entity_class=MatterBinarySensor,
+        required_attributes=(clusters.SmokeCoAlarm.Attributes.HardwareFaultAlert,),
+    ),
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=MatterBinarySensorEntityDescription(
+            key="SmokeCoAlarmSmokeStateSensor",
+            device_class=BinarySensorDeviceClass.SMOKE,
+            measurement_to_ha=lambda x: x
+            != clusters.SmokeCoAlarm.Enums.AlarmStateEnum.kNormal,
+        ),
+        entity_class=MatterBinarySensor,
+        required_attributes=(clusters.SmokeCoAlarm.Attributes.SmokeState,),
+    ),
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=MatterBinarySensorEntityDescription(
+            key="SmokeCoAlarmInterconnectSmokeAlarmSensor",
+            device_class=BinarySensorDeviceClass.SMOKE,
+            measurement_to_ha=lambda x: x
+            != clusters.SmokeCoAlarm.Enums.AlarmStateEnum.kNormal,
+            translation_key="interconnected_smoke_alarm",
+        ),
+        entity_class=MatterBinarySensor,
+        required_attributes=(clusters.SmokeCoAlarm.Attributes.InterconnectSmokeAlarm,),
+    ),
+    MatterDiscoverySchema(
+        platform=Platform.BINARY_SENSOR,
+        entity_description=MatterBinarySensorEntityDescription(
+            key="SmokeCoAlarmInterconnectCOAlarmSensor",
+            device_class=BinarySensorDeviceClass.SMOKE,
```

Comment:
Is this smoke or CO? because I think we have a device class for CO

Reply:
typo